### PR TITLE
Change substitution variables separator to `.`

### PR DESCRIPTION
**Pull request by sujuka99**

### CODE REVIEW

**Comment by raminqaf** [comment]
Why do we need to this?

**Comment by sujuka99** [comment]
Closing as it seems to be inconvenient for the users

**Comment by sujuka99** [comment]
reopening, the delimiter will be a dot instead.

**Review by disrupted — COMMENTED** [review]
please add an entry to the user migration guide

File: docs/docs/resources/variables/variable_substitution.yaml [hunk 1]
```diff
@@ -20,11 +20,11 @@
...
-      app_type: "${component_type}"
-      app_name: "${component_name}"
-      app_resources_requests_memory: "${component_app_resources_requests_memory}"
-      ${component_type}: "${component_app_labels_app_name}-${component_app_labels_app_type}"
-      test_placeholder_in_placeholder: "${component_app_labels_${component_type}}"
+      app_type: "${component.type}"
+      app_name: "${component.name}"
+      app_resources_requests_memory: "${component.app.resources.requests.memory}"
+      ${component.type}: "${component.app.labels.app_name}-${component.app.labels.app_type}"
+      test_placeholder_in_placeholder: "${component.app.labels.${component.type}}"
```
Comment (disrupted) [inline docs/docs/resources/variables/variable_substitution.yaml:27]: do we support this? 👀

Comment (sujuka99) [reply docs/docs/resources/variables/variable_substitution.yaml:27]: yes :grin:

File: kpops/utils/yaml_loading.py [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
-from string import Template
 from typing import Any
 
 import yaml
 from cachetools import cached
 from cachetools.keys import hashkey
 
+from kpops.utils.dict_ops import ImprovedTemplate
+
+# from kpops.utils.dict_ops import ImprovedTemplate
```
Comment (disrupted) [inline kpops/utils/yaml_loading.py:11]: please cleanup

File: kpops/utils/dict_ops.py [hunk 3]
```diff
@@ -96,3 +101,38 @@ def generate_substitution(
...
+            mapping = _ChainMap(kws, mapping)
+
+        # Helper function for .sub()
+        def convert(mo: re.Match):
+            named = mo.group("named") or mo.group("braced")
+            if named is not None:
+                try:
+                    if "." not in named:
+                        return str(mapping[named])
+                    return str(mapping[named.replace(".", "__")])
```
Comment (disrupted) [inline kpops/utils/dict_ops.py:128]: doesn't the `prepare_substitution` function already do the dot→dunder conversion?

Comment (sujuka99) [reply kpops/utils/dict_ops.py:128]: Update: Unsubstituted placeholders are left the same!

Comment (sujuka99) [reply kpops/utils/dict_ops.py:128]: I tried removing `prepare_substitution` and confirmed this

> This is for the input while prepare_substitution is for the substitution mapping.

**Review by disrupted — APPROVED** [review]

**Review by disrupted — APPROVED** [review]

**Review by disrupted — APPROVED** [review]

**Pull request merged**